src/pages/home/components/home-ranking/item/HomeRankingItem.tsx

```diff
@@ -0,0 +1,160 @@
+import { RankingWithJWTResponse } from '@/types/ranking';
+import * as S from './HomeRankingItem.style';
+import { useAuthStore } from '@/stores/auth.store';
+
+interface RankingItemProps {
+  data: RankingWithJWTResponse;
+  isJWT?: boolean;
+}
+
+const RankDisplay = ({
+  rank,
+  nickname,
+  totalTime,
+  isDayList,
+}: {
+  rank: number | undefined;
+  nickname: string | undefined;
+  totalTime: string | undefined;
+  isDayList?: boolean;
+}) => {
+  const { user } = useAuthStore();
+  const isCurrentUser = user?.nickname === nickname;
+
+  const renderRank = () => {
+    if (rank === 1) {
+      return (
+        <img
+          src="src/pages/home/components/home-ranking/item/1_medal.png"
+          alt="1st"
+        />
+      );
+    } else if (rank === 2) {
+      return (
+        <img
+          src="src/pages/home/components/home-ranking/item/2_medal.png"
+          alt="2nd"
+        />
+      );
+    } else if (rank === 3) {
+      return (
+        <img
+          src="src/pages/home/components/home-ranking/item/3_medal.png"
+          alt="3rd"
+        />
+      );
+    } else {
+      return rank;
+    }
+  };
+
+  return (
+    <S.RankItem isCurrentUser={isCurrentUser}>
+      <S.Rank isDayList={isDayList}>{renderRank()}</S.Rank>
+      <S.ItemWrap>
+        <S.NickName>{nickname}</S.NickName>
+        <S.TotalTime>{totalTime}</S.TotalTime>
+      </S.ItemWrap>
+    </S.RankItem>
+  );
+};
+
+function HomeRankingItem({ data, isJWT }: RankingItemProps) {
+  const {
+    userInfo,
+    prevPrevUserInfo,
+    prevUserInfo,
+    nextUserInfo,
+    nextNextUserInfo,
+  } = data;
+
+  const myListItems = [];
+
+  if (isJWT) {
```

Comment:
```suggestion
    const myListItems = [];

  if (isJWT && 'userInfo' in data) {
    const {
      userInfo,
      prevPrevUserInfo,
      prevUserInfo,
      nextUserInfo,
      nextNextUserInfo,
    } = data;
```

위처럼 `RankingWithJWTResponse ` 타입을 체크하고,
```
userInfo,
prevPrevUserInfo,
prevUserInfo,
nextUserInfo,
nextNextUserInfo,
```
를 사용할 수 있도록 해야할 것 같아요